src/ast.js

```diff
@@ -52,11 +52,11 @@ export default class Ast {
           if (!newNode && piece > 0 && node.value[piece - 1]) {
             // Element in sequence does not exist. It could have been empty
             // Let's provide the end of previous element
-            start = node.value[piece - 1].end_mark.pointer;
-            end = start + 1;
```

Comment:
Before we were simply guessing the end because we don't know much about the source map. This can be problematic becuase we may already be at end of file so adding one to position is incorrect, it can also be much harder for us to compute the next line/column because we'd have to traverse the source document to figure out if there is one.

I think it is simpler, safer and better to just provide the start index in this case.